src/aks-preview/azext_aks_preview/decorator.py

```diff
@@ -608,23 +655,29 @@ def _get_enable_pod_identity_with_kubenet(self, enable_validation: bool = False,
         """
         # read the original value passed by the command
         enable_pod_identity_with_kubenet = self.raw_param.get("enable_pod_identity_with_kubenet")
-        # try to read the property value corresponding to the parameter from the `mc` object
-        if (
-            self.mc and
-            self.mc.pod_identity_profile and
-            self.mc.pod_identity_profile.allow_network_plugin_kubenet is not None
-        ):
-            enable_pod_identity_with_kubenet = self.mc.pod_identity_profile.allow_network_plugin_kubenet
+        # In create mode, try to read the property value corresponding to the parameter from the `mc` object.
+        if self.decorator_mode == DecoratorMode.CREATE:
+            if (
+                self.mc and
+                self.mc.pod_identity_profile and
+                self.mc.pod_identity_profile.allow_network_plugin_kubenet is not None
+            ):
+                enable_pod_identity_with_kubenet = self.mc.pod_identity_profile.allow_network_plugin_kubenet
 
         # this parameter does not need dynamic completion
         # validation
         if enable_validation:
-            if self.mc and self.mc.network_profile and safe_lower(self.mc.network_profile.network_plugin) == "kubenet":
-                if not enable_pod_identity_with_kubenet and self._get_enable_pod_identity(enable_validation=False):
-                    raise RequiredArgumentMissingError(
-                        "--enable-pod-identity-with-kubenet is required for enabling pod identity addon "
-                        "when using Kubenet network plugin"
-                    )
+            if self.decorator_mode == DecoratorMode.CREATE:
+                if (
```

Comment:
nit: can we extract this kubenet validation with L608 to a helper method?

Reply:
Good point, working on a fix.

Reply:
Updated in commit [#eaeaa6b](https://github.com/Azure/azure-cli-extensions/pull/4243/commits/eaeaa6b408a905c9b25edbdc5f127ea3c15ed7f5).